provider/aws_test.go

```diff
@@ -713,6 +722,210 @@ func TestAWSApplyDryRun(t *testing.T) {
 	}
 }
 
+func TestAWSCreateRecordsCNAME(t *testing.T) {
+	provider := newAWSProvider(t, false)
+
+	_, err := provider.CreateZone("ext-dns-test.teapot.zalan.do.")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	records := []endpoint.Endpoint{{DNSName: "create-test.ext-dns-test.teapot.zalan.do.", Target: "foo.elb.amazonaws.com"}}
+
+	err = provider.CreateRecords("ext-dns-test.teapot.zalan.do.", records)
```

Comment:
It's untested, but so it is in all the other cases as well (A records and dryRun), but you see it nicely in the coverage report that the error case isn't tested.

However, these tests are a nightmare. I need to clean them up and would consider testing this case then.